docs/src/interfacer.md

```diff
@@ -1,55 +1,166 @@
+A component model should be implemented as a struct that is a concrete subtype
+of a `ComponentModelSimulation`. This struct should contain all of the
+information needed to run that simulation.
+
+Each `ComponentModelSimulation` must extend the following functions to be able
+to use our coupler. For some existing models, these are defined within
+ClimaCoupler.jl in that model’s `init.jl` file, but it is preferable
+for these to be defined in a model’s own repository. Note that the dispatch
+`::ComponentModelSimulation` in the function definitions given below should
+be replaced with the particular component model extending these functions.
+- `init`: construct and return an instance of the `ComponentModelSimulation`,
+and perform all initialization. This function should return a simulation that
+is ready to be stepped in the coupled simulation. This function currently
+doesn’t have a unified interface for arguments and may vary across component
+models, but this may change in the future.
+- `name(::ComponentModelSimulation)`: return a string containing the name of
+this `ComponentModelSimulation`, which is used for printing information about
+component models.
+- `step!(::ComponentModelSimulation, t)`: A function to update the
+simulation in-place with values calculate for time `t`. For the
+models we currently have implemented, this is a simple wrapper around
+the `step!` function implemented in SciMLBase.jl.
+- `reinit!(::ComponentModelSimulation)`: A function to restart a simulation
+after solving of the simulation has been paused or interrupted. Like
+`step!`, this is currently a simple wrapper around the `reinit!` function
+of SciMLBase.jl.
+- `get_model_state_vector(::ComponentModelSimulation)`: A function that
+returns the state vector of the simulation at its current state. This
+is used for checkpointing the simulation.
+
+### ComponentModelSimulation - optional functions
+- `update_sim!(::ComponentModelSimulation, csf, turbulent_fluxes)`: A
+function to update each of the fields of the component model simulation
+that are updated by the coupler. ClimaCoupler.jl provides defaults of
+this function for both `AtmosModelSimulation` and
+`SurfaceModelSimulation` that update each of the fields expected by
+the coupler. This function can optionally be extended to include
+additional field updates as desired.
+
+### AtmosModelSimulation - required functions
+In addition to the functions required for a general
+`ComponentModelSimulation`, an `AtmosModelSimulation` requires the
+following functions to retrieve and update its fields.
+- `get_field(::AtmosModelSimulation. ::Val{property})`: This getter
+function returns the value of the field property for the simulation
+in its current state. For an `AtmosModelSimulation`, it must be extended
+for the following properties:
+  - `air_density`: air density at the surface of the atmosphere
+  - `air_temperature`: air temperature at the surface of the atmosphere
+  - `energy`: TODO
```

Comment:
TODO?

Reply:
We should also include units of these required variables here.

Reply:
for the TODOs, I wasn't sure exactly what the quantities are (e.g. is energy column-integrated?). I'd appreciate your help filling them in, or we can go over them together :) 

I'm also not sure what units we use for these variables

Reply:
Sure I can add these in later on. :) Are you happy for me to push to this branch?

Reply:
Yes, that would be great! Thank you